src/fastoad/openmdao/tests/test_problem.py

```diff
@@ -115,6 +116,29 @@ def test_problem_read_inputs_before_setup(cleanup):
     assert_allclose(problem.get_val(name="z", units="m**2"), [4.0, 3.0])
     assert_allclose(problem["f"], 21.7572, atol=1.0e-4)
 
+    # Set values of inputs and outputs
+    # (FASTOADProblem has to defer output setting after setup to have it effective,
+    #  so this test works also for read after setup)
```

Comment:
Not to be too picky (at this pont I have to say it every PR) but when you say that FASTOADProbelm has to defer output setting after setup it is more that the only values that are setup before setup are values with shape_by_conn, no ?

Reply:
This is right. The point was to explain why I don't try to test also a `read_inputs()` operation after `setup()`.
I guess I could make it clearer...

Reply:
Actually, I realize the point of this comment was a bit different. That proves very well it was not clear enough. Modified in 4f93f70ea6f39a2d2224e2c10c2868151694b86c.
You'll tell me if it's clearer :)
BTW, I quickly modified a test component to have a `shape_by_conn=True` on an output. Maybe not the best test, on a one-element array, but it's a corner case, so it's better than nothing.

Reply:
Thank you for the added test.
As for the comment, I'm still 100% sure the second part is clear enough may I suggest:
_A FASTOADProblem will automatically set output values after setup using set_val(), which is functionally the same as doing the setup()->read_inputs() sequence._
May I also suggest that, between the problem setup and run_model on line 131 and 132 we assert that the value of variable such as g1 and g2 are not to their default ? Just to showcase what happens behind the scenes ?